src/main/java/com/ss/training/utopia/orchestrator/counter/security/AuthenticationFilter.java

```diff
@@ -0,0 +1,65 @@
+package com.ss.training.utopia.orchestrator.counter.security;
+
+import static com.auth0.jwt.algorithms.Algorithm.HMAC512;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.auth0.jwt.JWT;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+/**
+ * @author Justin O'Brien
+ */
+public class AuthenticationFilter extends UsernamePasswordAuthenticationFilter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuthenticationFilter.class);
+    private AuthenticationManager authenticationManager;
+
+    /**
+	 * @param authenticationManager
+	 */
+	public AuthenticationFilter(AuthenticationManager authenticationManager) {
+		this.authenticationManager = authenticationManager;
+    }
+    
+    @Override
+	public Authentication attemptAuthentication(HttpServletRequest request, HttpServletResponse response)
+			throws AuthenticationException {
+		UserCredentials userCredentials = null;
+		try {
+			userCredentials = new ObjectMapper().readValue(request.getInputStream(), UserCredentials.class);
+		} catch (Exception e) {
+			LOGGER.error("There was an error while attempting to extract user credentials from the request.\n"
+					+ e.toString());
+			return null;
+		}
+		UsernamePasswordAuthenticationToken authenticationToken = new UsernamePasswordAuthenticationToken(
+				userCredentials.getUsername(), userCredentials.getPassword(), new ArrayList<>());
+		return authenticationManager.authenticate(authenticationToken);
+	}
+
+    @Override
+	protected void successfulAuthentication(HttpServletRequest request, HttpServletResponse response, FilterChain chain,
+			Authentication authResult) throws IOException, ServletException {
+		UserDetailsClass userDetails = (UserDetailsClass) authResult.getPrincipal();
+		String token = JWT.create().withSubject(userDetails.getUsername())
+				.withExpiresAt(new Date(System.currentTimeMillis() + JwtProperties.EXPIRATION_TIME))
+				.sign(HMAC512(JwtProperties.SECRET.getBytes()));
```

Comment:
Akbar said we need to change this from HMAC512 to SHA256 in the other group's demo. Are you familiar with how that works?

Reply:
I'm not sure, but hopefully it'll just be something simple like importing a SHA256 method and using it in place of HMAC512, like we were talking about in our meeting today.